sklearn/decomposition/dict_learning.py

```diff
@@ -717,6 +717,19 @@ def dict_learning_online(X, n_components=2, alpha=1, n_iter=100,
         Number of iterations run. Returned only if `return_n_iter` is
         set to `True`.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from sklearn.decomposition import DictionaryLearning
+    >>> X = np.array([[0., 0., 1.],
+    ...               [1., 0., 0.],
+    ...               [2., 2., 2.],
+    ...               [2., 5., 4.]])
+    >>> dico = DictionaryLearning(n_components=2, alpha=1, random_state=1)
+    >>> dico.fit(X).components_
```

Comment:
Sorry, a little question. What is the purpose of `# doctest: +ELLIPSIS`.

Reply:
it says that `...` can be replaced by any string. The following lines replace the numbers with `...` so that in case of numerical instability, the test still passes. If the directive is not active, doctest would expect exactly that as the output, instead of accepting anything in place of `...`. Not sure if I explained it clearly. [reference](https://docs.python.org/3.7/library/doctest.html#doctest.ELLIPSIS)

Reply:
ahh yes! thanks!